otherlibs/unix/gethost.c

```diff
@@ -84,28 +85,36 @@ CAMLprim value caml_unix_gethostbyaddr(value a)
 {
   struct in_addr adr = GET_INET_ADDR(a);
   struct hostent * hp;
+  int addr_type = AF_INET;
+  socklen_t addr_len = 4;
+#if HAS_IPV6
+  if (caml_string_length(a) == 16) {
+    addr_type = AF_INET6;
+    addr_len = 16;
+  }
+#endif
 #if HAS_GETHOSTBYADDR_R == 7
   struct hostent h;
   char buffer[NETDB_BUFFER_SIZE];
   int h_errnop;
   caml_enter_blocking_section();
-  hp = gethostbyaddr_r((char *) &adr, 4, AF_INET,
+  hp = gethostbyaddr_r((char *) &adr, addr_len, addr_type,
                        &h, buffer, sizeof(buffer), &h_errnop);
   caml_leave_blocking_section();
 #elif HAS_GETHOSTBYADDR_R == 8
   struct hostent h;
   char buffer[NETDB_BUFFER_SIZE];
   int h_errnop, rc;
   caml_enter_blocking_section();
-  rc = gethostbyaddr_r((char *) &adr, 4, AF_INET,
+  rc = gethostbyaddr_r((char *) &adr, addr_len, addr_type,
                        &h, buffer, sizeof(buffer), &hp, &h_errnop);
   caml_leave_blocking_section();
   if (rc != 0) hp = NULL;
 #else
 #ifdef GETHOSTBYADDR_IS_REENTRANT
   caml_enter_blocking_section();
 #endif
-  hp = gethostbyaddr((char *) &adr, 4, AF_INET);
+  hp = gethostbyaddr((char *) &adr, addr_type, addr_type);
```

Comment:
Second parameter is wrong!
```suggestion
  hp = gethostbyaddr(adr, addr_len, addr_type);
```

Reply:
Oops :grimacing: